coffeecard/CoffeeCard.Library/Services/v2/AccountService.cs

```diff
@@ -210,6 +210,38 @@ public async Task UpdateUserGroup(UserGroup userGroup, int userId)
             await _context.SaveChangesAsync();
         }
 
+        public async Task<List<User>> SearchUsers(String search, int pageNum, int pageLength)
+        {
+            List<User> users = new List<User>();
+
+            int skip = pageNum * pageLength;
+
+            if (int.TryParse(search, out int id))
+            {
+                var user = await _context.Users.FirstOrDefaultAsync(u => u.Id == id);
+                if (user != null)
+                {
+                    users.Add(user);
+                    return users;
+                }
+            }
+
+            users = await _context.Users
+                .Where(u => u.Email.ToLower().StartsWith(search.ToLower()) || u.Name.ToLower().StartsWith(search.ToLower()))
+                .OrderBy(u => u.Id)
+                .Skip(skip).Take(pageLength)
+                .ToListAsync();
```

Comment:
Greatly reduce complexity of code and eliminate weird edge cases (you may think they're stupid, but they're important) with this change:

```suggestion
            users = await _context.Users
                .Where(u => u.Email.ToLower().StartsWith(search.ToLower()) || u.Name.ToLower().StartsWith(search.ToLower()) || u.Id.ToString().StartsWith(search))
                .OrderBy(u => u.Id)
                .Skip(skip).Take(pageLength)
                .ToListAsync();
```

Alternatively, to enforce an exact match on the ID:
```suggestion
            users = await _context.Users
                .Where(u => u.Email.ToLower().StartsWith(search.ToLower()) || u.Name.ToLower().StartsWith(search.ToLower()) || u.Id.ToString() == search
                .OrderBy(u => u.Id)
                .Skip(skip).Take(pageLength)
                .ToListAsync();
```